### PR TITLE
Introduce ResultsTable class

### PR DESCRIPTION
**Pull request by florian-huber**
Address #82

### CODE REVIEW

File: ms2query/ms2library.py [hunk 1]
```diff
@@ -235,23 +236,27 @@ def _get_analog_search_scores(self,
             spectrum_id = query_spectrum.get(
                 self.settings["spectrum_id_column_name"])
             query_spectrum_parent_mass = query_spectrum.get("parent_mass")
+            results_table = ResultsTable(preselection_cut_off=preselection_cut_off,
```
Comment (niekdejonge) [inline ms2query/ms2library.py:239]: Might be an option to pass on a selection of the ms2ds_scores already (the ones related to this query spectrum)

File: ms2query/ms2library.py [hunk 2]
```diff
@@ -235,23 +236,27 @@ def _get_analog_search_scores(self,
...
-            s2v_scores = self._get_s2v_scores(query_spectrum,
-                                              selected_spectrum_ids)
+            results_table.data["s2v_scores"] = self._get_s2v_scores(query_spectrum,
+                                                                    results_table.data.index.values)
+
+            parent_masses = np.array([self.parent_masses_library[x] for x in results_table.data.index])
+            results_table.add_parent_masses(parent_masses,
+                                            self.settings["base_nr_mass_similarity"])
 
-            dict_with_preselected_spectra_info[spectrum_id] = \
```
Comment (niekdejonge) [inline ms2query/ms2library.py:250]: select_and_normalize_scores_for_selected_spectra can be removed, since it is now replaced by ResultClass

Comment (niekdejonge) [reply ms2query/ms2library.py:250]: It cleans up the confusing functions and long variables really nicely!

File: ms2query/ms2library.py [hunk 3]
```diff
@@ -286,11 +291,15 @@ def _get_all_ms2ds_scores(self, query_spectra: List[Spectrum]
...
+            results_table: ResultsTable,
             ms2ds_scores: pd.Series,
-            preselection_cut_off: int
+            preselection_cut_off: int,
+            sort_on_average_ms2ds: bool = False
             ):
         """Returns preselected spectra and average and closely related ms2ds
 
+        results_table:
+            RestultsTable object to collect scores and data about spectra of interest.
```
Comment (niekdejonge) [inline ms2query/ms2library.py:302]: Typo RestultsTable

File: ms2query/ms2library.py [hunk 4]
```diff
@@ -13,6 +13,7 @@
     get_parent_mass, get_inchikey_information
 from ms2query.utils import load_pickled_file
```
Comment (niekdejonge) [inline ms2query/ms2library.py:14]: I think this was much needed! :) Makes it a lot clearer and better structured.

File: ms2query/ms2library.py [hunk 5]
```diff
@@ -235,23 +236,27 @@ def _get_analog_search_scores(self,
...
 
             # Select the library spectra that have the highest ms2ds with the
             # query spectrum and calculate the average and related ms2ds scores
-            selected_spectrum_ids, average_ms2ds_scores, \
-                closely_related_inchikey_scores = \
+            results_table = \
                 self._calculate_averages_and_preselection(
+                    results_table,
                     ms2ds_scores[spectrum_id],
                     preselection_cut_off)
```
[the remark this answers is not in the document]
Comment (florian-huber) [reply ms2query/ms2library.py]: OK, great.

File: ms2query/results_table.py [hunk 6]
```diff
@@ -47,3 +49,29 @@ def add_parent_masses(self, parent_masses, base_nr_mass_similarity):
...
+
+        Args:
+        ------
+        predictions:
+            An iterable containing the ms2query model meta scores
+        """
+        self.data["ms2query_model_prediction"] = predictions
+        self.data.sort_values(by=["ms2query_model_prediction"],
+                              ascending=False,
+                              inplace=True)
```
Comment (florian-huber) [inline ms2query/results_table.py:72]: The sorting is fine I guess. But I think we should add a test for this (maybe build a unit test for `ResultsTable` anyway).
Can be done later as reaction to #97.

**Pull request merged**